corehq/apps/app_manager/tests/test_tasks.py

```diff
@@ -0,0 +1,122 @@
+        build2.save()
+        prune_auto_generated_builds(self.domain, app.id)
+        build_ids = get_build_ids(app.domain, app.id)
+        self.assertEqual(len(build_ids), 2)
+        self.assertNotIn(build2.id, build_ids)
+
+    def test_dependencies_feature_removed(self):
+        factory = AppFactory(build_version='2.40.0')
+        m0, f0 = factory.new_basic_module('register', 'case')
+        f0.source = get_simple_form(xmlns=f0.unique_id)
+        factory.app.profile = {'features': {'dependencies': ['coffee']}}
+        factory.app.save()
+        self.addCleanup(factory.app.delete)
+        build1 = factory.app.make_build()
+        build1.save()
+
+        factory.app.profile = {'features': {'dependencies': []}}
+        factory.app.save()
+
+        with patch("corehq.apps.app_manager.tasks.metrics_counter") as metric_counter_mock:
+            make_app_build(factory.app, "comment", user_id="user_id")
+            metric_counter_mock.assert_called_with('commcare.app_build.dependencies_removed')
+
+    def test_dependencies_feature_metrics_not_triggerd(self):
+        factory = AppFactory(build_version='2.40.0')
+        m0, f0 = factory.new_basic_module('register', 'case')
+        f0.source = get_simple_form(xmlns=f0.unique_id)
+        factory.app.profile = {'features': {'dependencies': []}}
+        factory.app.save()
+        self.addCleanup(factory.app.delete)
+        build1 = factory.app.make_build()
+        build1.save()
+
+        factory.app.save()
+
+        with patch("corehq.apps.app_manager.tasks.metrics_counter") as metric_counter_mock:
+            make_app_build(factory.app, "comment", user_id="user_id")
+            metric_counter_mock.assert_not_called()
+
+    def test_refresh_data_dictionary_from_app(self):
+        factory = AppFactory(build_version='2.56.0')
+        m0, f0 = factory.new_basic_module('update', 'case')
+        factory.form_requires_case(f0, update={'texture': '/data/texture'})
+        f0.source = get_simple_form(xmlns=f0.unique_id)
+        factory.form_uses_usercase(f0, update={
+            'favorite_color': ConditionalCaseUpdate(question_path='/data/favorite_color')
+        })
+        m1, f1 = factory.new_advanced_module('advanced', 'case')
+        factory.form_requires_case(f1, 'person')
+        factory.app.save()
+        self.addCleanup(factory.app.delete)
+
+        with patch('corehq.apps.app_manager.tasks.get_app') as get_app:
+            get_app.return_value = factory.app
+            refresh_data_dictionary_from_app(self.domain, factory.app.get_id)
+
+        types = CaseType.objects.filter(domain=self.domain)
+        self.assertEqual(types.count(), 3)
+        self.assertSetEqual(set([t.name for t in types]), set(['case', 'person', 'commcare-user']))
+
+        props = CaseProperty.objects.filter(case_type__domain=self.domain)
+        self.assertEqual(props.count(), 2)
+        self.assertSetEqual(set([p.name for p in props]), set(['texture', 'favorite_color']))
```

Comment:
nit
```suggestion
        self.assertEqual({t.name for t in types}, {'case', 'person', 'commcare-user'})

        props = CaseProperty.objects.filter(case_type__domain=self.domain)
        self.assertEqual(props.count(), 2)
        self.assertEqual({p.name for p in props}, {'texture', 'favorite_color'})
```

Reply:
huh, why does `assertSetEqual` even exist?

Reply:
Looks like it [could be useful when you're comparing a `set` with a `frozenset`](https://stackoverflow.com/questions/75251966/assertequal-vs-assertsetequlal-in-unittest), although I'm not sure why `assertEqual` doesn't call it for that scenario too.